spock-core/src/main/java/org/spockframework/runtime/PlatformParameterizedSpecRunner.java

```diff
@@ -40,6 +40,7 @@ void runParameterizedFeature(SpockExecutionContext context, ParameterizedFeature
       return;
     }
 
+    context = context.withChildStoreProvider();
```

Comment:
I am not a big fan of overriding parameter values in the method.

Reply:
I understand, but we've used this pattern for the context so far. We could change it in another PR.